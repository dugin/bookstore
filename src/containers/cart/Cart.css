.Cart {
  min-height: calc(100vh - 140px);
  margin-top: 20px;
  position: relative; }
  @media (min-width: 768px) {
    .Cart .col-md-3 {
      flex: 0 0 30%;
      max-width: 30%; } }
  .Cart__empty {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    font-size: 20px; }
    @media (min-width: 768px) {
      .Cart__empty {
        font-size: 28px; } }
    .Cart__empty .material-icons {
      font-size: 100px;
      color: #000; }
  .Cart__summary {
    padding-top: 10px;
    border: 1px solid #d1d1d1;
    background-color: #fff;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: 14px;
    margin-top: -1px;
    max-height: 200px; }
    .Cart__summary__price, .Cart__summary__discount, .Cart__summary__total {
      margin-bottom: 15px; }
    .Cart__summary__price__title, .Cart__summary__total__title {
      font-weight: bold; }
    .Cart__summary__discount__title, .Cart__summary__discount__value {
      color: #939393; }
    .Cart__summary__total {
      font-size: 22px; }
    .Cart__summary__discount__value {
      text-decoration: line-through; }
    .Cart__summary__price__value, .Cart__summary__discount__value, .Cart__summary__total__value {
      text-align: end; }
