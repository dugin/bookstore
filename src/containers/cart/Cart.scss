@import "../../variables";
.Cart{

  background-color: $color--white;
}