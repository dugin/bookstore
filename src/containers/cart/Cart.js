import React, {Component} from 'react';
import './Cart.css';
import CartBook from "../cartBook/CartBook";
import {connect} from 'react-redux';
import * as utils from '../../utils/booksUtils';
import NumberFormat from 'react-number-format';

export class Cart extends Component {

    render() {
        const books = this.props.books;

        return (
            <div className="Cart container">
                {books && books.length > 0 ? (
                    <div className=" row">
                        <section className="col-md-8 col-12">
                            <CartBook/>
                        </section>
                        <section className="col-md-3 ml-auto col-12 Cart__summary">

                            {utils.setDiscountSum(books) !== 0 && (
                                <div>
                                    <div className="Cart__summary__price row">
                                        <p className="col-5 Cart__summary__price__title">Preço sem Desconto</p>

                                        <NumberFormat className="col-7 Cart__summary__price__value"
                                                      value={utils.setPriceSum(books)}
                                                      decimalPrecision={2} displayType={'text'} decimalSeparator=','
                                                      prefix={'R$ '}/>
                                    </div>

                                    <div className="Cart__summary__discount row">
                                        <p className="col-5 Cart__summary__discount__title">Desconto</p>
                                        <NumberFormat className="col-7 Cart__summary__discount__value"
                                                      value={utils.setDiscountSum(books)}
                                                      decimalPrecision={2} displayType={'text'} decimalSeparator=','
                                                      prefix={'R$ '}/>
                                    </div>
                                </div>
                            )}

                            <div className="Cart__summary__total row">
                                <p className="col-4 Cart__summary__total__title">Total</p>
                                <NumberFormat className="col-8 Cart__summary__total__value pl-0"
                                              value={utils.setDiscountedPriceSum(books)}
                                              decimalPrecision={2} displayType={'text'} decimalSeparator=','
                                              prefix={'R$ '}/>
                            </div>
                        </section>
                    </div>
                ) : (
                    <section className="Cart__empty">
                        <i className="material-icons">shopping_cart</i>
                        <p className="Cart__empty__message">Seu carrinho está vazio.</p>

                    </section>
                )}
            </div>
        )
    }

}

export const mapStateToProps = (state, props) => ({books: state.books});


export default connect(mapStateToProps, null)(Cart);