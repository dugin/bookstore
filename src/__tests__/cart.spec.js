import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Cart from "../containers/cart/Cart";

describe('Cart Page', () => {

    const cart = shallow(<Cart/>);

    it('renders properly', () => {
        expect(toJson(cart)).toMatchSnapshot();
    });

});