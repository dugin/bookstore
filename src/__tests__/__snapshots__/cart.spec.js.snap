// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Page renders properly 1`] = `
<div
  className="Cart"
/>
`;
